src/utils/types/dialog_types.rs

```diff
@@ -0,0 +1,12 @@
+use super::ContentActionType;
+use leptos::{html::Dialog, NodeRef};
+
+#[derive(Clone, Default)]
+pub struct ReportModalData {
+  pub id: i32,
+  pub content_type: ContentActionType,
+  pub creator_actor_id: String,
```

Comment:
Enum

Reply:
I don't think it's needed here. In this case, it's always going to be the id of a user.